tests/system/Helpers/HTMLHelperTest.php

```diff
@@ -248,6 +248,27 @@ public function testScriptTagWithIndexpage()
         $this->assertSame($expected, script_tag($target, true));
     }
 
+    public function testScriptTagWithSrc()
+    {
+        $target   = ['src' => 'http://site.com/js/mystyles.js'];
+        $expected = '<script src="http://site.com/js/mystyles.js" type="text/javascript"></script>';
+        $this->assertSame($expected, script_tag($target));
+    }
+
+    public function testScriptTagWithSrcWithoutProtocol()
+    {
+        $target   = ['src' => 'js/mystyles.js'];
+        $expected = '<script src="http://example.com/js/mystyles.js" type="text/javascript"></script>';
+        $this->assertSame($expected, script_tag($target));
+    }
+
+    public function testScriptTagWithSrcAndAttributes()
+    {
+        $target   = ['src' => 'js/mystyles.js', 'charset' => 'UTF-8', 'defer' => '', 'async' => null];
```

Comment:
is `'defer' => ''` intended? i assume this should be `null`?

Reply:
Oh, probably it is intended to test the behavior of the value `''`,
but `defer=""` is invalid.

Reply:
Hello,

yes, `'defer' => ''` is indeed intended to test the behaviour of the value `''`. But note that `defer=""` is a valid writing of the attribute:
* https://dev.w3.org/html5/spec-LC/common-microsyntaxes.html#boolean-attributes
* https://www.w3.org/TR/html401/intro/sgmltut.html#h-3.3.4.2

Given my understanding, all the following writings are valid and equivalent:
```
<script src="..." defer></script>
<script src="..." defer=""></script>
<script src="..." defer="defer"></script>
```

Doc updated using the W3 "minimized form" terminology, and I added a test on calling `script_tag()` without any argument.

Reply:
https://html.spec.whatwg.org/#boolean-attributes
Yes, @xlii-chl is correct.